File: FluentFTP.CSharpExamples/QuickStart.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {

	internal static class QuickStart {

		public static void Example() {

			// create an FTP client and specify the host, username and password
			// (delete the credentials to use the "anonymous" account)
			var client = new FtpClient("123.123.123.123", "david", "pass123");

			// connect to the server and automatically detect working FTP settings
			client.AutoConnect();

			// get a list of files and directories in the "/htdocs" folder
			foreach (FtpListItem item in client.GetListing("/htdocs")) {

				// if this is a file
				if (item.Type == FtpObjectType.File) {

					// get the file size
					long size = client.GetFileSize(item.FullName);

					// calculate a hash for the file on the server side (default algorithm)
					FtpHash hash = client.GetChecksum(item.FullName);
				}

				// get modified date/time of the file or folder
				DateTime time = client.GetModifiedTime(item.FullName);

			}

			// upload a file
			client.UploadFile(@"C:\MyVideo.mp4", "/htdocs/MyVideo.mp4");

			// move the uploaded file
			client.MoveFile("/htdocs/MyVideo.mp4", "/htdocs/MyVideo_2.mp4");

			// download the file again
			client.DownloadFile(@"C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4");

			// compare the downloaded file with the server
			if (client.CompareFile(@"C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4") == FtpCompareResult.Equal) { }

			// delete the file
			client.DeleteFile("/htdocs/MyVideo_2.mp4");

			// upload a folder and all its files
			client.UploadDirectory(@"C:\website\videos\", @"/public_html/videos", FtpFolderSyncMode.Update);

			// upload a folder and all its files, and delete extra files on the server
			client.UploadDirectory(@"C:\website\assets\", @"/public_html/assets", FtpFolderSyncMode.Mirror);

			// download a folder and all its files
			client.DownloadDirectory(@"C:\website\logs\", @"/public_html/logs", FtpFolderSyncMode.Update);

			// download a folder and all its files, and delete extra files on disk
			client.DownloadDirectory(@"C:\website\dailybackup\", @"/public_html/", FtpFolderSyncMode.Mirror);

			// delete a folder recursively
			client.DeleteDirectory("/htdocs/extras/");

			// check if a file exists
			if (client.FileExists("/htdocs/big2.txt")) { }

			// check if a folder exists
			if (client.DirectoryExists("/htdocs/extras/")) { }

			// upload a file and retry 3 times before giving up
			client.Config.RetryAttempts = 3;
			client.UploadFile(@"C:\MyVideo.mp4", "/htdocs/big.txt", FtpRemoteExists.Overwrite, false, FtpVerify.Retry);

			// disconnect! good bye!
			client.Disconnect();

		}

		public static async Task ExampleAsync() {

			// create an FTP client and specify the host, username and password
			// (delete the credentials to use the "anonymous" account)
			var client = new AsyncFtpClient("123.123.123.123", "david", "pass123");

			// connect to the server and automatically detect working FTP settings
			await client.AutoConnect();

			// get a list of files and directories in the "/htdocs" folder
			foreach (FtpListItem item in await client.GetListing("/htdocs")) {

				// if this is a file
				if (item.Type == FtpObjectType.File) {

					// get the file size
					long size = await client.GetFileSize(item.FullName);

					// calculate a hash for the file on the server side (default algorithm)
					FtpHash hash = await client.GetChecksum(item.FullName);
				}

				// get modified date/time of the file or folder
				DateTime time = await client.GetModifiedTime(item.FullName);
			}

			// upload a file
			await client.UploadFile(@"C:\MyVideo.mp4", "/htdocs/MyVideo.mp4");

			// move the uploaded file
			await client.MoveFile("/htdocs/MyVideo.mp4", "/htdocs/MyVideo_2.mp4");

			// download the file again
			await client.DownloadFile(@"C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4");

			// compare the downloaded file with the server
			if (await client.CompareFile(@"C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4") == FtpCompareResult.Equal) { }

			// delete the file
			await client.DeleteFile("/htdocs/MyVideo_2.mp4");

			// upload a folder and all its files
			await client.UploadDirectory(@"C:\website\videos\", @"/public_html/videos", FtpFolderSyncMode.Update);

			// upload a folder and all its files, and delete extra files on the server
			await client.UploadDirectory(@"C:\website\assets\", @"/public_html/assets", FtpFolderSyncMode.Mirror);

			// download a folder and all its files
			await client.DownloadDirectory(@"C:\website\logs\", @"/public_html/logs", FtpFolderSyncMode.Update);

			// download a folder and all its files, and delete extra files on disk
			await client.DownloadDirectory(@"C:\website\dailybackup\", @"/public_html/", FtpFolderSyncMode.Mirror);

			// delete a folder recursively
			await client.DeleteDirectory("/htdocs/extras/");

			// check if a file exists
			if (await client.FileExists("/htdocs/big2.txt")) { }

			// check if a folder exists
			if (await client.DirectoryExists("/htdocs/extras/")) { }

			// upload a file and retry 3 times before giving up
			client.Config.RetryAttempts = 3;
			await client.UploadFile(@"C:\MyVideo.mp4", "/htdocs/big.txt", FtpRemoteExists.Overwrite, false, FtpVerify.Retry);

			// disconnect! good bye!
			await client.Disconnect();
		}

	}
}

File: FluentFTP.VBExamples/QuickStart.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module QuickStart
		Sub Example()
			Dim client = New FtpClient("123.123.123.123", "david", "pass123")
			client.AutoConnect()

			For Each item As FtpListItem In client.GetListing("/htdocs")

				If item.Type = FtpObjectType.File Then
					Dim size As Long = client.GetFileSize(item.FullName)
					Dim hash As FtpHash = client.GetChecksum(item.FullName)
				End If

				Dim time As DateTime = client.GetModifiedTime(item.FullName)
			Next

			client.UploadFile("C:\MyVideo.mp4", "/htdocs/MyVideo.mp4")
			client.MoveFile("/htdocs/MyVideo.mp4", "/htdocs/MyVideo_2.mp4")
			client.DownloadFile("C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4")

			If client.CompareFile("C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4") = FtpCompareResult.Equal Then
			End If

			client.DeleteFile("/htdocs/MyVideo_2.mp4")
			client.UploadDirectory("C:\website\videos\", "/public_html/videos", FtpFolderSyncMode.Update)
			client.UploadDirectory("C:\website\assets\", "/public_html/assets", FtpFolderSyncMode.Mirror)
			client.DownloadDirectory("C:\website\logs\", "/public_html/logs", FtpFolderSyncMode.Update)
			client.DownloadDirectory("C:\website\dailybackup\", "/public_html/", FtpFolderSyncMode.Mirror)
			client.DeleteDirectory("/htdocs/extras/")

			If client.FileExists("/htdocs/big2.txt") Then
			End If

			If client.DirectoryExists("/htdocs/extras/") Then
			End If

			client.Config.RetryAttempts = 3
			client.UploadFile("C:\MyVideo.mp4", "/htdocs/big.txt", FtpRemoteExists.Overwrite, False, FtpVerify.Retry)
			client.Disconnect()
		End Sub

		Async Function ExampleAsync() As Task
			Dim client = New AsyncFtpClient("123.123.123.123", "david", "pass123")
			Await client.AutoConnect()

			For Each item As FtpListItem In Await client.GetListing("/htdocs")

				If item.Type = FtpObjectType.File Then
					Dim size As Long = Await client.GetFileSize(item.FullName)
					Dim hash As FtpHash = Await client.GetChecksum(item.FullName)
				End If

				Dim time As DateTime = Await client.GetModifiedTime(item.FullName)
			Next

			Await client.UploadFile("C:\MyVideo.mp4", "/htdocs/MyVideo.mp4")
			Await client.MoveFile("/htdocs/MyVideo.mp4", "/htdocs/MyVideo_2.mp4")
			Await client.DownloadFile("C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4")

			If Await client.CompareFile("C:\MyVideo_2.mp4", "/htdocs/MyVideo_2.mp4") = FtpCompareResult.Equal Then
			End If

			Await client.DeleteFile("/htdocs/MyVideo_2.mp4")
			Await client.UploadDirectory("C:\website\videos\", "/public_html/videos", FtpFolderSyncMode.Update)
			Await client.UploadDirectory("C:\website\assets\", "/public_html/assets", FtpFolderSyncMode.Mirror)
			Await client.DownloadDirectory("C:\website\logs\", "/public_html/logs", FtpFolderSyncMode.Update)
			Await client.DownloadDirectory("C:\website\dailybackup\", "/public_html/", FtpFolderSyncMode.Mirror)
			Await client.DeleteDirectory("/htdocs/extras/")

			If Await client.FileExists("/htdocs/big2.txt") Then
			End If

			If Await client.DirectoryExists("/htdocs/extras/") Then
			End If

			client.Config.RetryAttempts = 3
			Await client.UploadFile("C:\MyVideo.mp4", "/htdocs/big.txt", FtpRemoteExists.Overwrite, False, FtpVerify.Retry)
			Await client.Disconnect()
		End Function
	End Module
End Namespace
